testsuite/Array4DTest/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+#find_package (Boost COMPONENTS system filesystem unit_test_framework REQUIRED)
+#include_directories(${Boost_INCLUDE_DIRS})
+add_definitions (-DBOOST_TEST_DYN_LINK)
+add_executable(Array4DTest Array4DTest.cpp)
+add_test(Array4DTest ${CMAKE_CURRENT_SOURCE_DIR}/Array4DTest)
```

Comment:
That is the problem, travis does an out of source build (`mkdir build && cd build && ...`), so you should you use `${CMAKE_CURRENT_BINARY_DIR}` instead of `${CMAKE_CURRENT_SOURCE_DIR}`.